### PR TITLE
Ataya integration

### PR DESCRIPTION
**Pull request by tony19911010**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Review by ilappe — CHANGES_REQUESTED** [review]
Thanks @tony19911010 for this contribution!
i reviewed the code etc and please see my comments
@melamedbn will review the remain stuff (playbook etc.) 

in addition please check the:
- [secret error](https://app.circleci.com/pipelines/github/demisto/content/313492/workflows/81b17d0a-a082-44a9-b6b0-350bbc80c0a0/jobs/726923?invite=true#step-118-269_75)
- [validation errors](https://app.circleci.com/pipelines/github/demisto/content/313492/workflows/81b17d0a-a082-44a9-b6b0-350bbc80c0a0/jobs/726923?invite=true#step-119-15637_92)

File: Packs/Ataya/Integrations/Ataya/Ataya_test.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+MOCK_IMSI = "46666610000001"
+
+MOCK_ASSIGN_USER_RESPONSE = {
+    "status": "assigned",
+    "resources": [MOCK_IMSI]
+}
+
+
+def test_assign_command(requests_mock):
```
[the remark this answers is not in the document]
Comment (kgal-pan) [reply Packs/Ataya/Integrations/Ataya/Ataya_test.py:13]: Check out the [coverage report](https://output.circle-artifacts.com/output/job/80c33046-fd5b-47cb-95c7-43807ce0e0b2/artifacts/0/artifacts/coverage_report/html/d_fb61a98ad6ba4b52_Ataya_py.html) to see what lines are not being hit by the unit tests..

File: Packs/Ataya/Integrations/Ataya/Ataya.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    if userStatus == 'unassigned':
+        raise DemistoException('Assign User Fail', res=response)
+
+    return f'User {imsi} {userStatus}'
+
+
+''' MAIN FUNCTION '''
+
+
+def main() -> None:
```
Comment (kgal-pan) [inline Packs/Ataya/Integrations/Ataya/Ataya.py:78]: No need to run coverage on the `main` function:

```suggestion
def main() -> None:  # pragma: no cover
```

**Comment by ilappe** [comment]
looks good for me
waiting for @melamedbn review

**Comment by kgal-pan** [comment]
[Commit job failing because `.gitlab` has changed.](https://github.com/demisto/content/actions/runs/6701825354/job/18221790698?pr=30511#step:8:100).

**Comment by tony19911010** [comment]
> [Commit job failing because `.gitlab` has changed.](https://github.com/demisto/content/actions/runs/6701825354/job/18221790698?pr=30511#step:8:100).

Already updated

**Comment by tony19911010** [comment]
Already updated

**Comment by melamedbn** [comment]
Hello @tony19911010,

I’ve been reviewing the content you’ve provided and wanted to share some feedback to further refine and enhance the materials. Your efforts are very much appreciated and with a few tweaks, I believe we can perfect this submission.

### Incident Fields:
- **Field Utilization**: Could we leverage the existing incident fields to prevent duplication? For instance, instead of creating 'Ataya IP', using 'Source/Destination/Device IP' would be more aligned with our standards. If you integrate existing fields, please ensure this is reflected across all relevant content items.
- Please make all fields unsearchable and remove them for the pack-ignore.

### Mapper:
- **Terminology Consistency**: For the mapper, let’s stick to the established naming conventions: consider renaming to 'Ataya - Incoming Mapper'.

### Playbooks: Ataya - Securely Logging Device Access to Network
- **Descriptive Clarity**: A more detailed title and description will greatly improve understanding of the playbook’s purpose.
- **Visual Order**: Aligning the grid arrows will not only enhance the visual appeal but also contribute to readability.
- **Task Titling**: Strive for more descriptive task titles; for example, 'Is Ataya Enabled?' is more informative than the generic 'isIntegrationAvailable'.
- **Flow Explanation**: The current playbook flow could use additional comments to aid in comprehension.
- **Integration Checks**: Ensure the PAN-OS integration check is performed at the outset and consider using 'skipifunavailable' to avoid making PAN-OS a mandatory dependency.
- **Task Redundancy**: Tasks 3 and 7 appear identical; could one be removed or are there distinguishing factors that aren't apparent?
- **Task Relevance**: In Task 6, can you elaborate on how 'removing the user ID mapping' ties into the cmd argument provided?
- **Task Descriptions**: A pass through all tasks to flesh out descriptions will significantly aid users in following the playbook logic.
- **Command Clarity**: The 'ataya-assign-user' command should be accompanied by a more comprehensive explanation. Additionally, the discrepancy between Task 11’s title and the command functionality needs to be clarified.
- **Approval Process**: The data collection task should have a more explicit question. What is the approval seeking to confirm?

### Layout:
- **Grid Optimization**: To maximize space and improve layout flow, please rearrange the sections as per the suggested state in the provided screenshots.

image

image


Your collaboration is key to our continuous improvement, and these suggestions aim to align your valuable contribution with our best practices. If there's anything you’d like to discuss or require further details on, I’m here to help.

Best regards,
Ben

**Review by ilappe — APPROVED** [review]

**Comment by tony19911010** [comment]
Hello @melamedbn,
Thanks for the review.

About the **Incident Fields** and **Mapper**, I have already update the related files according to the comments you provided above.

About the **Playbooks**, I also update the some of the description and title and also slightly modify the flow on Playbook to meet the requirement.

Please review it again and see if there has something need to be modified.

**Comment by melamedbn** [comment]
Hi @tony19911010,

I hope this message finds you well. First off, I want to say a huge thanks for your diligent work on implementing the feedback. The progress is clear and appreciated! There are a few additional details that I believe could enhance the playbook even further:

### Event Type Inspection for Session Control (Task 4):

* Can you please add an 'Else' path?
* Also, it seems like 'Update User ID' (Task 3) and 'Create User ID' (Task 7) are overlapping in functionality. To streamline, perhaps we can merge these actions into a single task, governed by a condition that checks incident.eventtype for both 'create' and 'modify'.

### Task Printing (Task 1):

* Is there a specific need for the print function here? If it's for debugging purposes, do remember to remove it before finalizing.

### Clarity in Data Collection (Task 10):

* Under the message tab, the subject line could be more descriptive to clearly convey the required action for data collection.

Best regards,
Ben

**Comment by tony19911010** [comment]
Hello @melamedbn,
Thanks again for your effort on reviewing the PR. Your feedback is helpful for more clarify of our integration.
I've been update our playbook based on all of your feedbacks above.

Please check this update again, appreciate your help very much.

**Comment by melamedbn** [comment]
Hi @tony19911010,

Great job! You just need to add an ELSE path to task 8 and we are good to go.

Ben

**Comment by tony19911010** [comment]
Hi @melamedbn,
I've already added the ELSE path, and also update the playbook image. Please check

**Review by melamedbn — APPROVED** [review]

**Comment by ilappe** [comment]
Hi @tony19911010
great job!
please share (via Slack or put here a link to shared drive)
a demo video.
the demo video should include the following:
- upload the pack to XSOAR and show how it's look in the market place, (you can do it with `demisto-sdk upload -z -x --insecure -i Packs/Ataya`
- configure the integration with valid/invalid credentials and show test module result
- run the command in the playground and show the context output

**Comment by tony19911010** [comment]
Hi @ilappe,
I have completed the demo video, and I just put the video link here.

https://atayalan-my.sharepoint.com/:f:/g/personal/tony_ataya_io/EvW_dzYlqfJPlJJOzCA109sBD0fiDML6zkd2lLkttcMj3g?e=MHXsuJ

The demo videos are separate to four parts.
1. Pack overview on SOAR.
2. Create and Remove the api token credentials on Ataya Harmony web, and check if the Integration test_module can pass the test or not.
3. Run the integration command in the playground, and check the client status on Ataya Harmony web can sccessfully change the status from non-assigned to assigned.
4. Generate the incidents manually, and check the incidents and playbook execution.

**Comment by ilappe** [comment]
Thanks @tony19911010 
great job and demo!

**Pull request merged**